include/xsparse/util/container_traits.hpp

```diff
@@ -22,6 +22,18 @@ namespace xsparse::util
 
         template <class Key, class Val>
         using Map = TMap<Key, Val>;
+
```

Comment:
Just static_assert for the presence of these functions rather than adding an abstract api.